### PR TITLE
Improve feature tests

### PR DESCRIPTION
**Pull request by axelboc**
First batch of commits to bring the feature tests improvements made in the React 18 branch #1119. I've made separate commits for clarity. Another PR to come.

### CODE REVIEW

File: packages/app/src/__tests__/DomainSlider.test.tsx [hunk 1]
```diff
@@ -172,30 +166,14 @@ test('clamp domain in symlog scale', async () => {
...
-  await user.keyboard('{ArrowLeft}');
+  await user.type(maxThumb, '{ArrowLeft}');
   expect(maxInput).toHaveValue('5.40006e+301');
   expect(maxThumb).toHaveAttribute('aria-valuenow', '99'); // does not jump back to 81
 
   // Remove focus now to avoid state update after unmount
   maxThumb.blur();
 });
 
-test('show min/max autoscale toggles in tooltip (pressed by default)', async () => {
```
Comment (axelboc) [inline packages/app/src/__tests__/DomainSlider.test.tsx:184]: Already covered by other tests.

File: packages/app/src/__tests__/DomainSlider.test.tsx [hunk 2]
```diff
@@ -68,32 +68,28 @@ test('update domain when moving thumbs (with keyboard)', async () => {
   const maxInput = screen.getByLabelText('max');
 
   // Move min thumb one step to the left with keyboard
-  minThumb.focus();
-  await user.keyboard('{ArrowLeft}');
+  await user.type(minThumb, '{ArrowLeft}');
```
Comment (axelboc) [inline packages/app/src/__tests__/DomainSlider.test.tsx:71]: `type` gives focus to the element first (amongst other things).

File: cypress/e2e/app.cy.ts [hunk 3]
```diff
@@ -158,9 +158,8 @@ describe('/mock', () => {
     cy.selectExplorerNode('fourD');
 
     cy.findByRole('figure', { name: 'fourD' }).as('vis').should('be.visible');
-    cy.findAllByRole('slider', { name: 'Dimension slider' })
-      .should('have.length', 2)
-      .last()
+    cy.findAllByRole('slider', { name: /D[0-9]/u }).should('have.length', 2);
+    cy.findAllByRole('slider', { name: 'D1' })
```
Comment (axelboc) [inline cypress/e2e/app.cy.ts:162]: Gave the domain slider thumbs more specific names so it's easier to find the one we want

File: packages/app/src/__tests__/NexusPack.test.tsx [hunk 4]
```diff
@@ -453,55 +453,16 @@ test('retry fetching automatically when selecting other NxSpectrum slice', async
...
-  // Let fetch of first slice succeed
-  jest.runAllTimers();
-  await expect(screen.findByRole('figure')).resolves.toBeVisible();
-  d0Slider.blur(); // remove focus to avoid state update after unmount
-
-  jest.runOnlyPendingTimers();
-  jest.useRealTimers();
-});
-
-test('retry fetching supporting datasets automatically when selecting other NxImage slice', async () => {
```
Comment (axelboc) [inline packages/app/src/__tests__/NexusPack.test.tsx:478]: Too similar to the previous test -- it doesn't bring anything more.

**Review by loichuder — APPROVED** [review]
Nice :slightly_smiling_face:

**Pull request merged**